pyro/infer/autoguide/guides.py

```diff
@@ -1419,20 +1419,25 @@ def get_deltas(self, save_params=None):
                 aux_value = aux_value + dep(aux_values[upstream])
             aux_values[name] = aux_value
 
-            # Shift by loc, reshape, and transform to constrained space.
-            unconstrained = aux_value + loc
-            shape = self._unconstrained_shapes[name]
-            if torch._C._get_tracing_state() or unconstrained.shape != shape:
-                sample_shape = unconstrained.shape[:-1]
-                unconstrained = unconstrained.reshape(sample_shape + shape)
+            # Shift by loc and reshape.
+            batch_shape = torch.broadcast_shapes(
+                aux_value.shape[:-1], self._batch_shapes[name]
+            )
+            unconstrained = (aux_value + loc).reshape(
+                batch_shape + self._unconstrained_event_shapes[name]
+            )
+            if not is_identically_zero(log_density):
+                log_density = log_density.reshape(batch_shape + (-1,)).sum(-1)
+
+            # Transform to constrained space.
             transform = biject_to(site["fn"].support)
             value = transform(unconstrained)
-
-            # Create a Delta distribution.
             if compute_density and conditional != "delta":
-                ldj = transform.inv.log_abs_det_jacobian(value, unconstrained)
-                ldj = sum_rightmost(ldj, ldj.dim() - value.dim() + site["fn"].event_dim)
```

Comment:
This is no longer necessary now that we have `IndependentTransform` and that all distributions satisfy `fn.event_dim == fn.support.event_dim`.  In fact the code earlier in this file from which this code is copied is obsolete, but I have avoided cluttering this PR with unrelated changes.

To be safe I've added an assertion in case a user has created a custom distribution with invalid shape metadata.  I've also added a shape test with `LKJCholesky` which would have failed before our work on `IndependentTransform`.